front/lib/api/data_sources.ts

```diff
@@ -73,8 +76,30 @@ export async function deleteDataSource(
     });
   }
 
-  const dustAPIProjectId = dataSource.dustAPIProjectId;
+  await dataSource.delete(auth, { transaction, hardDelete: false });
```

Comment:
why not use softDetete?

Reply:
Seen IRL. common interface is `.delete` which requires an explicit `hardDelete` parameter when the model supports soft deletion.